File: SGP_Web/Scripts/SGP/Reportes.js
$(document).ready(function () {

    google.charts.load('current', { 'packages': ['corechart', 'line'] });

    var app = new Vue({
        el: '#app',
        data: {
            dProgramacionFacturacion: [],
        },
        methods: {
            ListarProgramacionFacturacion: function () {
                var data = {
                    anio: $('#txtAnio').val()
                };

                var table = null;
                var clearTable = $('#TbReporte').DataTable();
                clearTable.clear();
                $('#TbReporte').empty();
                clearTable.destroy();

                axios.post('/Reportes/ProgramacionFacturacion', data).then(response => {
                    var listData = response.data;
                    if (listData.length > 0) {
                        $('#FilaTabla').removeClass('d-none');
                        $('#ReporteGrafico').addClass('d-none');
                        $('.btnImprimirGrafico').addClass('d-none');

                        var argTitulo = $('#txtAnio').val();

                        if ($.fn.dataTable.isDataTable('#TbReporte')) {
                            table = $('#TbReporte').DataTable();
                        }
                        else {
                            var item = 0;
                            table = $('#TbReporte').DataTable({
                                "searching": false,
                                "processing": true,
                                "paging": false,
                                "ordering": false,
                                "info": listData.length < 7 ? false : true,
                                "data": listData,
                                "columns": [
                                //{ 'data': 'co_proyecto', 'width': '0%' },
                                {
                                    "render": function (data, type, row, meta) {
                                        return parseInt(meta["row"]) + 1;
                                    }
                                },
                                { 'data': 'Proyecto', 'width': '5%' },
                                { 'data': 'Enero', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Febrero', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Marzo', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Abril', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Mayo', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Junio', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Julio', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Agosto', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Setiembre', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Octubre', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Noviembre', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'Diciembre', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                { 'data': 'total', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                ],
                                "language": {

                                    "sProcessing": "Procesando...",
                                    "sLengthMenu": "",
                                    "sZeroRecords": "No se encontraron resultados",
                                    "sEmptyTable": "Ningún dato disponible en esta tabla",
                                    "sInfo": "Mostrando registros del _START_ al _END_ de un total de _TOTAL_ registros",
                                    "sInfoEmpty": "Mostrando registros del 0 al 0 de un total de 0 registros",
                                    "sInfoFiltered": "(filtrado de un total de _MAX_ registros)",
                                    "sInfoPostFix": "",
                                    "sSearch": "Ingrese palabra clave para filtrar:",
                                    "sUrl": "",
                                    "sInfoThousands": ",",
                                    "sLoadingRecords": "Cargando...",
                                    "oPaginate": {
                                        "sFirst": "Primero",
                                        "sLast": "Último",
                                        "sNext": "Siguiente",
                                        "sPrevious": "Anterior"
                                    },
                                    "oAria": {
                                        "sSortAscending": ": Activar para ordenar la columna de manera ascendente",
                                        "sSortDescending": ": Activar para ordenar la columna de manera descendente"
                                    }
                                },
                                dom: 'Bfrtip',
                                buttons: [
                                    {
                                        extend: 'excel',
                                        title: 'Programación de la Facturación ' + argTitulo,
                                        text: '<i class="fa fa-file-excel-o fa-1x"> Excel</i>',
                                        className: 'btn btn-success'
                                    }
                                    , {
                                        extend: 'print',
                                        title: 'Programación de la Facturación ' + argTitulo,
                                        text: '<i class="fa fa-file-text-o fa-1x"> Imprimir</i>',
                                        className: 'btn btn-secondary'
                                    }
                                ]
                            });
                        }
                    } else {
                        this.clearDatatable();
                    }


                    //this.dProgramacionFacturacion = response.data;
                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
            ListarFacturacionMes: function () {
                var data = {
                    anio: $('#txtAnio').val()
                };
                axios.post('/Reportes/FacturacionMes', data).then(response => {


                    if (response.data.length > 0) {
                        $('#FilaTabla').addClass('d-none');
                        $('#ReporteGrafico').removeClass('d-none');
                        $('.btnImprimirGrafico').removeClass('d-none');
                        var arr = response.data;

                        var argTitulo = $('#txtAnio').val();

                        var objFacturacionMes = [];
                        var SimboloMoneda = arr[0]["SimboloMoneda"];
                        for (var i = 0; i < arr.length; i++) {
                            var subObjFacturacionMes = [];
                            subObjFacturacionMes.push(arr[i]["NombreMes"]);
                            subObjFacturacionMes.push(arr[i]["Importe"]);
                            subObjFacturacionMes.push(arr[i]["SimboloMoneda"] + arr[i]["Importe"]);
                            objFacturacionMes.push(subObjFacturacionMes);
                        }

                        var dataTable = new google.visualization.DataTable();
                        dataTable.addColumn('string', 'Mes');
                        dataTable.addColumn('number', SimboloMoneda);

                        dataTable.addColumn({ type: 'string', role: 'tooltip' });
                        dataTable.addRows(objFacturacionMes);

                        var options = {
                            title: 'Facturación por Mes ' + argTitulo,
                            vAxis: { format: 'decimal' },
                            width: 900,
                            height: 500
                        };
                        var chart = new google.visualization.ColumnChart(document.getElementById('ReporteGrafico'));
                        chart.draw(dataTable, options);

                    } else {
                        $('#ReporteGrafico').addClass('d-none');
                        $('.btnImprimirGrafico').addClass('d-none');
                    }

                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
            ListarCrecimientoProgramacionFacturacion: function () {

                var data = {
                    anio: $('#txtAnio').val()
                };
                axios.post('/Reportes/FacturacionMes', data).then(response => {

                    if (response.data.length > 0) {
                        $('#FilaTabla').addClass('d-none');
                        $('#ReporteGrafico').removeClass('d-none');
                        $('.btnImprimirGrafico').removeClass('d-none');

                        var argTitulo = $('#txtAnio').val();
                        var arr = response.data;
                        var obj = [];
                        var acumuladorImporte = 0;
                        for (var i = 0; i < arr.length; i++) {
                            var subOBJ = [];
                            subOBJ.push(arr[i]["NombreMes"]);
                            subOBJ.push(arr[i]["Importe"]);
                            acumuladorImporte += arr[i]["Importe"];
                            subOBJ.push(acumuladorImporte);
                            obj.push(subOBJ);
                        }

                        var data = new google.visualization.DataTable();
                        data.addColumn('string', 'Mes');
                        data.addColumn('number', 'Facturación X Mes');
                        data.addColumn('number', 'Crecimiento de Venta ' + $('#txtAnio').val());

                        data.addRows(obj);

                        var options = {
                            title: 'Crecimiento Facturación por Mes ' + argTitulo,
                            colors: ['#a52714', '#097138'],
                            vAxis: { format: 'decimal' },
                            axes: {
                                x: {
                                    0: { side: 'top' }
                                }
                            }
                        };

                        var chart = new google.visualization.LineChart(document.getElementById('ReporteGrafico'));
                        chart.draw(data, options);

                    } else {
                        $('#ReporteGrafico').addClass('d-none');
                        $('.btnImprimirGrafico').addClass('d-none');
                    }

                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
            ImprimirGrafico: function () {

                var printContents = document.getElementById('ReporteGraficos').innerHTML;
                w = window.open();
                w.document.write(printContents);
                w.document.close(); // necessary for IE >= 10
                w.focus(); // necessary for IE >= 10
                w.print();
                w.close();
                return true;
            },
        },
        created: function () {
            //this.ListarProgramacionFacturacion();
        },
    });

    var FacturacionMes = new Vue({
        el: '#FacturacionMes',
        data: {
            dFacturacionMes: [],
        },
        methods: {
            ListarFacturacionMes: function () {

                var data = {
                    anio: $('#txtAnio').val()
                };
                axios.post('/Reportes/FacturacionMes', data).then(response => {


                    if (response.data.length > 0) {
                        $('#ReporteGrafico').removeClass('d-none');
                        var arr = response.data;
                        var objFacturacionMes = [];
                        var SimboloMoneda = arr[0]["SimboloMoneda"];
                        for (var i = 0; i < arr.length; i++) {
                            var subObjFacturacionMes = [];
                            subObjFacturacionMes.push(arr[i]["NombreMes"]);
                            subObjFacturacionMes.push(arr[i]["Importe"]);
                            subObjFacturacionMes.push(arr[i]["SimboloMoneda"] + arr[i]["Importe"]);
                            objFacturacionMes.push(subObjFacturacionMes);
                        }

                        var dataTable = new google.visualization.DataTable();
                        dataTable.addColumn('string', 'Mes');
                        dataTable.addColumn('number', SimboloMoneda);

                        dataTable.addColumn({ type: 'string', role: 'tooltip' });
                        dataTable.addRows(objFacturacionMes);

                        var options = {
                            chart: {
                                title: '',
                                subtitle: '',
                            },
                            vAxis: { format: 'decimal' },
                            width: 900,
                            height: 500
                        };
                        var chart = new google.visualization.ColumnChart(document.getElementById('ReporteGrafico'));
                        chart.draw(dataTable, options);

                    } else {
                        $('#ReporteGrafico').addClass('d-none');
                    }

                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
        },
        created: function () {
            //this.ListarFacturacionMes();
        },
    });

    var CrecimientoFacturacionMes = new Vue({
        el: '#CrecimientoFacturacionMes',
        data: {
            dProgramacionFacturacion: [],
        },
        methods: {
            ListarCrecimientoProgramacionFacturacion: function () {

                var data = {
                    anio: $('#txtAnio').val()
                };
                axios.post('/Reportes/FacturacionMes', data).then(response => {


                    if (response.data.length > 0) {
                        $('#ReporteGrafico').removeClass('d-none');
                        var arr = response.data;
                        var obj = [];
                        var acumuladorImporte = 0;
                        for (var i = 0; i < arr.length; i++) {
                            var subOBJ = [];
                            subOBJ.push(arr[i]["NombreMes"]);
                            subOBJ.push(arr[i]["Importe"]);
                            acumuladorImporte += arr[i]["Importe"];
                            subOBJ.push(acumuladorImporte);
                            obj.push(subOBJ);
                        }

                        var data = new google.visualization.DataTable();
                        data.addColumn('string', 'Mes');
                        data.addColumn('number', 'Facturación X Mes');
                        data.addColumn('number', 'Crecimiento de Venta ' + $('#txtAnio').val());

                        data.addRows(obj);

                        var options = {
                            colors: ['#a52714', '#097138'],
                            vAxis: { format: 'decimal' },
                        };

                        var chart = new google.visualization.LineChart(document.getElementById('ReporteGrafico'));
                        chart.draw(data, options);

                    } else {
                        $('#ReporteGrafico').addClass('d-none');
                    }

                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },

        },
        created: function () {
            //this.ListarCrecimientoProgramacionFacturacion();
        },
    });

    var ConsultaEjecucionPago = new Vue({
        el: '#ConsultaEjecucionPago',
        data: {
            dProgramacionFacturacion: [],
        },
        methods: {
            ListarCronogramaEjecucionPago: function () {
                var data = {
                    so_interna: $('#txtSolicitudInterna').val(),
                    nu_ordencompra: $('#txtOrdenCompra').val(),
                    nu_recepcion: $('#txtNroRecepcion').val(),
                    nu_facturacion: $('#txtFactura').val(),
                };

                var table = null;
                var clearTable = $('#TbReporteEjecucionPago').DataTable();
                clearTable.clear();
                $('#TbReporteEjecucionPago').empty();
                clearTable.destroy();

                axios.post('/Reportes/ConsultaEjecucionPago', data).then(response => {
                    var listData = response.data;
                    if (listData.length > 0) {
                        
                        if ($.fn.dataTable.isDataTable('#TbReporteEjecucionPago')) {
                            table = $('#TbReporteEjecucionPago').DataTable();
                        }
                        else {
                            var item = 0;
                            table = $('#TbReporteEjecucionPago').DataTable({
                                "searching": false,
                                "processing": true,
                                "paging": false,
                                "ordering": false,
                                "info": listData.length < 7 ? false : true,
                                "data": listData,
                                "columns": [
                                { 'data': 'de_proyecto', 'width': '5%' },
                                { 'data': 'co_SRT', 'width': '5%' },
                                { 'data': 'so_interna', 'width': '5%' },
                                { 'data': 'nu_ordencompra', 'width': '5%' },
                                { 'data': 'nu_recepcion', 'width': '5%' },
                                { 'data': 'nu_facturacion', 'width': '5%' },
                                { 'data': 'mo_importefacturacion', 'width': '5%', render: $.fn.dataTable.render.number(',', '.', 2, '') },
                                ],
                                "language": {

                                    "sProcessing": "Procesando...",
                                    "sLengthMenu": "",
                                    "sZeroRecords": "No se encontraron resultados",
                                    "sEmptyTable": "Ningún dato disponible en esta tabla",
                                    "sInfo": "Mostrando registros del _START_ al _END_ de un total de _TOTAL_ registros",
                                    "sInfoEmpty": "Mostrando registros del 0 al 0 de un total de 0 registros",
                                    "sInfoFiltered": "(filtrado de un total de _MAX_ registros)",
                                    "sInfoPostFix": "",
                                    "sSearch": "Ingrese palabra clave para filtrar:",
                                    "sUrl": "",
                                    "sInfoThousands": ",",
                                    "sLoadingRecords": "Cargando...",
                                    "oPaginate": {
                                        "sFirst": "Primero",
                                        "sLast": "Último",
                                        "sNext": "Siguiente",
                                        "sPrevious": "Anterior"
                                    },
                                    "oAria": {
                                        "sSortAscending": ": Activar para ordenar la columna de manera ascendente",
                                        "sSortDescending": ": Activar para ordenar la columna de manera descendente"
                                    }
                                },
                                dom: 'Bfrtip',
                                buttons: [
                                    {
                                        extend: 'excel',
                                        title: "Reporte Ejecución de Pago ",
                                        text: '<i class="fa fa-file-excel-o fa-1x"> Excel</i>',
                                        className: 'btn btn-success'
                                    }
                                    , {

                                        extend: 'print',
                                        title: "Reporte Ejecución de Pago " ,
                                        text: '<i class="fa fa-file-text-o fa-1x"> Imprimir</i>',
                                        className: 'btn btn-secondary'
                                    }
                                ]
                            });
                        }
                    } else {
                        this.clearDatatable();
                    }


                    //this.dProgramacionFacturacion = response.data;
                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
        },
        created: function () {
            //this.ListarCronogramaEjecucionPago();
        },
    });

    var ConsultaProgramacionPago = new Vue({
        el: '#ConsultaProgramacionPago',
        data: {
            dEstadoCronograma: [],
            dProgramacionPago: [],
            dResponsable: [],
        },
        methods: {
            ListarProgramacionPago: function () {
                var data = {
                    st_cronograma: $('#cboEstadoCronograma').val(),
                    fe_pago: $('#txtFechaInicioBusqueda').val(),
                    co_responsable: $('#cboResponsable').val(),
                };

                var table = null;
                var clearTable = $('#TbReporteProgramacionPago').DataTable();
                clearTable.clear();
                $('#TbReporteProgramacionPago').empty();
                clearTable.destroy();

                axios.post('/Reportes/ConsultaProgramacionPago', data).then(response => {
                    var listData = response.data;
                    if (listData.length > 0) {

                        if ($.fn.dataTable.isDataTable('#TbReporteProgramacionPago')) {
                            table = $('#TbReporteProgramacionPago').DataTable();
                        }
                        else {
                            var item = 0;
                            table = $('#TbReporteProgramacionPago').DataTable({
                                "searching": false,
                                "processing": true,
                                "paging": false,
                                "ordering": false,
                                "info": listData.length < 7 ? false : true,
                                "data": listData,
                                "columns": [
                                { 'data': 'de_proyecto', 'width': '35%' },
                                { 'data': 'co_SRT', 'width': '15%' },
                                { 'data': 'fe_pago', 'width': '5%' },
                                { 'data': 'fe_programada', 'width': '5%' },
                                { 'data': 'nu_hito', 'width': '5%' },
                                { 'data': 'de_hito', 'width': '15%' },
                                {
                                    'data': 'SimboloMoneda', 'width': '10%',
                                   "render": function (data, type, row, meta) {
                                       return '<h6> ' + row["SimboloMoneda"] + ' ' + ConvertirMiles(row["mo_importe"]) + ' </h6>';
                                   }
                                },
                                { 'data': 'Responsable', 'width': '15%' },                                
                                ],
                                "language": {

                                    "sProcessing": "Procesando...",
                                    "sLengthMenu": "",
                                    "sZeroRecords": "No se encontraron resultados",
                                    "sEmptyTable": "Ningún dato disponible en esta tabla",
                                    "sInfo": "Mostrando registros del _START_ al _END_ de un total de _TOTAL_ registros",
                                    "sInfoEmpty": "Mostrando registros del 0 al 0 de un total de 0 registros",
                                    "sInfoFiltered": "(filtrado de un total de _MAX_ registros)",
                                    "sInfoPostFix": "",
                                    "sSearch": "Ingrese palabra clave para filtrar:",
                                    "sUrl": "",
                                    "sInfoThousands": ",",
                                    "sLoadingRecords": "Cargando...",
                                    "oPaginate": {
                                        "sFirst": "Primero",
                                        "sLast": "Último",
                                        "sNext": "Siguiente",
                                        "sPrevious": "Anterior"
                                    },
                                    "oAria": {
                                        "sSortAscending": ": Activar para ordenar la columna de manera ascendente",
                                        "sSortDescending": ": Activar para ordenar la columna de manera descendente"
                                    }
                                },
                                dom: 'Bfrtip',
                                buttons: [
                                    {
                                        extend: 'excel',
                                        title: "Reporte Programación de Pago ",
                                        text: '<i class="fa fa-file-excel-o fa-1x"> Excel</i>',
                                        className: 'btn btn-success'
                                    }
                                    , {

                                        extend: 'print',
                                        title: "Reporte Programación de Pago ",
                                        text: '<i class="fa fa-file-text-o fa-1x"> Imprimir</i>',
                                        className: 'btn btn-secondary'
                                    }
                                ]
                            });
                        }
                    } else {
                        this.clearDatatable();
                    }


                    //this.dProgramacionFacturacion = response.data;
                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
            ListarEstadosCronograma: function () {
                var data = {
                    co_tabla: 10,
                };
                axios.post('/TGeneral/CargaTGeneral', data).then(response => {
                    this.dEstadoCronograma = response.data;
                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
            ListarResponsable: function () {
                var data = {
                    st_responsable: 1,
                };
                axios.post('/Responsable/CargaGrilla', data).then(response => {
                    this.dResponsable = response.data;
                }).catch(error => {
                    console.log(error);
                    this.errored = true;
                });
            },
        },
        created: function () {
            this.ListarEstadosCronograma();
            this.ListarResponsable();
        },
    });

    var ConvertirMiles = function (n) {
        n = n.toString()
        while (true) {
            var n2 = n.replace(/(\d)(\d{3})($|,|\.)/g, '$1,$2$3')
            if (n == n2) break
            n = n2
        }
        return n
    }

    $('#txtFechaInicioBusqueda').datepicker({
        format: 'dd/mm/yyyy',
        autoclose: true,
        language: "es"
    }).on('hide', function (selected) {
        //app.Listartabla();
    });

});

File: SGP_Data/Reportes.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SGP_Data
{
    public class Reportes
    {
        private readonly static Reportes _instance = new Reportes();

        private Reportes(){}

        public static Reportes Instance
        {
            get
            {
                return _instance;
            }
        }

        public List<SGP_Entity.Reportes.ProgramacionFacturacion> Sp_Sel_ProgramacionFacturacion(SGP_Entity.Reportes C)
        {
            using (SqlConnection con = new SqlConnection(ConfigurationManager.ConnectionStrings["cnx"].ConnectionString))
            {
                con.Open();
                using (SqlCommand com = new SqlCommand("Sp_Sel_ProgramacionFacturacion", con))
                {
                    com.CommandType = CommandType.StoredProcedure;
                    com.Parameters.Add("@argAnio", SqlDbType.Int).Value = C.anio;

                    List<SGP_Entity.Reportes.ProgramacionFacturacion> list = new List<SGP_Entity.Reportes.ProgramacionFacturacion>();
                    using (IDataReader dataReader = com.ExecuteReader())
                    {

                        while (dataReader.Read())
                        {

                            SGP_Entity.Reportes.ProgramacionFacturacion obj = new SGP_Entity.Reportes.ProgramacionFacturacion();

                            if (dataReader["Proyecto"] != DBNull.Value) { obj.Proyecto = (string)dataReader["Proyecto"]; }
                            if (dataReader["Enero"] != DBNull.Value) { obj.Enero = dataReader["Enero"].ToString(); }
                            if (dataReader["Febrero"] != DBNull.Value) { obj.Febrero = dataReader["Febrero"].ToString(); }
                            if (dataReader["Marzo"] != DBNull.Value) { obj.Marzo = dataReader["Marzo"].ToString(); }
                            if (dataReader["Abril"] != DBNull.Value) { obj.Abril = dataReader["Abril"].ToString(); }
                            if (dataReader["Mayo"] != DBNull.Value) { obj.Mayo = dataReader["Mayo"].ToString(); }
                            if (dataReader["Junio"] != DBNull.Value) { obj.Junio = dataReader["Junio"].ToString(); }
                            if (dataReader["Julio"] != DBNull.Value) { obj.Julio = dataReader["Julio"].ToString(); }
                            if (dataReader["Agosto"] != DBNull.Value) { obj.Agosto = dataReader["Agosto"].ToString(); }
                            if (dataReader["Septiembre"] != DBNull.Value) { obj.Setiembre = dataReader["Septiembre"].ToString(); }
                            if (dataReader["Octubre"] != DBNull.Value) { obj.Octubre = dataReader["Octubre"].ToString(); }
                            if (dataReader["Noviembre"] != DBNull.Value) { obj.Noviembre = dataReader["Noviembre"].ToString(); }
                            if (dataReader["Diciembre"] != DBNull.Value) { obj.Diciembre = dataReader["Diciembre"].ToString(); }
                            if (dataReader["total"] != DBNull.Value) { obj.total = dataReader["Total"].ToString(); }

                            list.Add(obj);

                        }
                    }
                    return list;
                }

            }
        }

        public List<SGP_Entity.Reportes.FacturacionMes> Sp_Sel_FacturacionMes(SGP_Entity.Reportes C)
        {
            using (SqlConnection con = new SqlConnection(ConfigurationManager.ConnectionStrings["cnx"].ConnectionString))
            {
                con.Open();
                using (SqlCommand com = new SqlCommand("Sp_Sel_FacturacionMes", con))
                {
                    com.CommandType = CommandType.StoredProcedure;
                    com.Parameters.Add("@argAnio", SqlDbType.Int).Value = C.anio;

                    List<SGP_Entity.Reportes.FacturacionMes> list = new List<SGP_Entity.Reportes.FacturacionMes>();
                    using (IDataReader dataReader = com.ExecuteReader())
                    {

                        while (dataReader.Read())
                        {

                            SGP_Entity.Reportes.FacturacionMes obj = new SGP_Entity.Reportes.FacturacionMes();

                            if (dataReader["NombreMes"] != DBNull.Value) { obj.NombreMes = (string)dataReader["NombreMes"]; }
                            if (dataReader["Importe"] != DBNull.Value) { obj.Importe = (decimal)dataReader["Importe"]; }
                            if (dataReader["SimboloMoneda"] != DBNull.Value) { obj.SimboloMoneda = (string)dataReader["SimboloMoneda"]; }
                            list.Add(obj);

                        }
                    }
                    return list;
                }

            }
        }

        public List<SGP_Entity.Reportes.ProgramacionFacturacion> Sp_Sel_ComparativoAnio(SGP_Entity.Reportes C)
        {
            using (SqlConnection con = new SqlConnection(ConfigurationManager.ConnectionStrings["cnx"].ConnectionString))
            {
                con.Open();
                using (SqlCommand com = new SqlCommand("Sp_Sel_ComparativoAnual", con))
                {
                    com.CommandType = CommandType.StoredProcedure;
                    com.Parameters.Add("@argAnio", SqlDbType.Int).Value = C.anio;
                    com.Parameters.Add("@argAnioAnterior", SqlDbType.Int).Value = C.anioAnterior;

                    List<SGP_Entity.Reportes.ProgramacionFacturacion> list = new List<SGP_Entity.Reportes.ProgramacionFacturacion>();
                    using (IDataReader dataReader = com.ExecuteReader())
                    {

                        while (dataReader.Read())
                        {

                            SGP_Entity.Reportes.ProgramacionFacturacion obj = new SGP_Entity.Reportes.ProgramacionFacturacion();

                            if (dataReader["Anio"] != DBNull.Value) { obj.Proyecto = dataReader["Anio"].ToString(); }
                            if (dataReader["1"] != DBNull.Value) { obj.Enero = dataReader["1"].ToString(); }
                            if (dataReader["2"] != DBNull.Value) { obj.Febrero = dataReader["2"].ToString(); }
                            if (dataReader["3"] != DBNull.Value) { obj.Marzo = dataReader["3"].ToString(); }
                            if (dataReader["4"] != DBNull.Value) { obj.Abril = dataReader["4"].ToString(); }
                            if (dataReader["5"] != DBNull.Value) { obj.Mayo = dataReader["5"].ToString(); }
                            if (dataReader["6"] != DBNull.Value) { obj.Junio = dataReader["6"].ToString(); }
                            if (dataReader["7"] != DBNull.Value) { obj.Julio = dataReader["7"].ToString(); }
                            if (dataReader["8"] != DBNull.Value) { obj.Agosto = dataReader["8"].ToString(); }
                            if (dataReader["9"] != DBNull.Value) { obj.Setiembre = dataReader["9"].ToString(); }
                            if (dataReader["10"] != DBNull.Value) { obj.Octubre = dataReader["10"].ToString(); }
                            if (dataReader["11"] != DBNull.Value) { obj.Noviembre = dataReader["11"].ToString(); }
                            if (dataReader["12"] != DBNull.Value) { obj.Diciembre = dataReader["12"].ToString(); }
                            list.Add(obj);

                        }
                    }
                    return list;
                }

            }
        }

        public List<SGP_Entity.Reportes.ConsultaEjecucionPago> Sp_Sel_ConsultaEjecucionPago(SGP_Entity.Reportes.ConsultaEjecucionPago C)
        {
            using (SqlConnection con = new SqlConnection(ConfigurationManager.ConnectionStrings["cnx"].ConnectionString))
            {
                con.Open();
                using (SqlCommand com = new SqlCommand("Sp_Sel_ConsultaEjecucionPago", con))
                {
                    com.CommandType = CommandType.StoredProcedure;
                    com.Parameters.Add("@so_interna", SqlDbType.VarChar,30).Value = C.so_interna;
                    com.Parameters.Add("@nu_ordencompra", SqlDbType.VarChar, 20).Value = C.nu_ordencompra;
                    com.Parameters.Add("@nu_recepcion", SqlDbType.VarChar, 20).Value = C.nu_recepcion;
                    com.Parameters.Add("@nu_facturacion", SqlDbType.VarChar,50).Value = C.nu_facturacion;

                    List<SGP_Entity.Reportes.ConsultaEjecucionPago> list = new List<SGP_Entity.Reportes.ConsultaEjecucionPago>();
                    using (IDataReader dataReader = com.ExecuteReader())
                    {
                        while (dataReader.Read())
                        {
                            SGP_Entity.Reportes.ConsultaEjecucionPago obj = new SGP_Entity.Reportes.ConsultaEjecucionPago();

                            if (dataReader["so_interna"] != DBNull.Value) { obj.so_interna = dataReader["so_interna"].ToString(); }
                            if (dataReader["nu_ordencompra"] != DBNull.Value) { obj.nu_ordencompra = dataReader["nu_ordencompra"].ToString(); }
                            if (dataReader["nu_recepcion"] != DBNull.Value) { obj.nu_recepcion = dataReader["nu_recepcion"].ToString(); }
                            if (dataReader["nu_facturacion"] != DBNull.Value) { obj.nu_facturacion = dataReader["nu_facturacion"].ToString(); }
                            if (dataReader["mo_importefacturacion"] != DBNull.Value) { obj.mo_importefacturacion = (decimal)dataReader["mo_importefacturacion"]; }
                            if (dataReader["co_SRT"] != DBNull.Value) { obj.co_SRT = dataReader["co_SRT"].ToString(); }
                            if (dataReader["de_proyecto"] != DBNull.Value) { obj.de_proyecto = dataReader["de_proyecto"].ToString(); }
                            //if (dataReader["fe_pago"] != DBNull.Value) { obj.fe_pago = dataReader["fe_pago"].ToString(); }
                            list.Add(obj);
                        }
                    }
                    return list;
                }

            }
        }


        public List<SGP_Entity.Reportes.ConsultaProgramacionPago> Sp_Sel_ConsultaProgramacionPago(SGP_Entity.Reportes.ConsultaProgramacionPago C)
        {
            using (SqlConnection con = new SqlConnection(ConfigurationManager.ConnectionStrings["cnx"].ConnectionString))
            {
                con.Open();
                using (SqlCommand com = new SqlCommand("Sp_Sel_ConsultaProgramacionPago", con))
                {
                    com.CommandType = CommandType.StoredProcedure;
                    com.Parameters.Add("@st_cronograma", SqlDbType.VarChar, 30).Value = C.st_cronograma;
                    com.Parameters.Add("@fe_pago", SqlDbType.DateTime).Value = C.fe_pago;
                    com.Parameters.Add("@co_responsable", SqlDbType.VarChar, 30).Value = C.co_responsable;

                    List<SGP_Entity.Reportes.ConsultaProgramacionPago> list = new List<SGP_Entity.Reportes.ConsultaProgramacionPago>();
                    using (IDataReader dataReader = com.ExecuteReader())
                    {
                        while (dataReader.Read())
                        {
                            SGP_Entity.Reportes.ConsultaProgramacionPago obj = new SGP_Entity.Reportes.ConsultaProgramacionPago();

                            if (dataReader["id_cronograma"] != DBNull.Value) { obj.id_cronograma = dataReader["id_cronograma"].ToString(); }
                            if (dataReader["de_proyecto"] != DBNull.Value) { obj.de_proyecto = dataReader["de_proyecto"].ToString(); }
                            if (dataReader["co_SRT"] != DBNull.Value) { obj.co_SRT = dataReader["co_SRT"].ToString(); }
                            if (dataReader["SimboloMoneda"] != DBNull.Value) { obj.SimboloMoneda = dataReader["SimboloMoneda"].ToString(); }
                            if (dataReader["mo_importe"] != DBNull.Value) { obj.mo_importe = (decimal)dataReader["mo_importe"]; }
                            if (dataReader["nu_hito"] != DBNull.Value) { obj.nu_hito = dataReader["nu_hito"].ToString(); }
                            if (dataReader["de_hito"] != DBNull.Value) { obj.de_hito = dataReader["de_hito"].ToString(); }
                            if (dataReader["Responsable"] != DBNull.Value) { obj.Responsable = dataReader["Responsable"].ToString(); }
                            if (dataReader["fe_pago"] != DBNull.Value) { obj.fe_pago = (dataReader["fe_pago"].ToString() != "" ? dataReader["fe_pago"].ToString().Substring(0, 10) : ""); }
                            if (dataReader["fe_programada"] != DBNull.Value) { obj.fe_programada = (dataReader["fe_programada"].ToString() != "" ? dataReader["fe_programada"].ToString().Substring(0, 10) : ""); }
                            list.Add(obj);
                        }
                    }
                    return list;
                }

            }
        }




    }
}
